pkg/issuer/venafi/client/venaficlient.go

```diff
@@ -183,16 +190,44 @@ func configForIssuer(iss cmapi.GenericIssuer, secretsLister internalinformers.Se
 			Credentials: &endpoint.Authentication{
 				APIKey: apiKey,
 			},
+			Client: httpClientForVcert(&httpClientForVcertOptions{
+				UserAgent: ptr.To(userAgent),
+			}),
```

Comment:
Previously we did not supply a custom `http.Client` when connecting to Venafi Cloud,
instead relying on the default created by the vcert-sdk.

Here I've refactored and re-used the `httpClientForVcert` function which had previously only existed for the purpose of configuring the TLS renegotation setting when connecting to TPP.

Now it also allows us to configure the User-Agent header.